src/spacegame3/gamedata/time/StarDateFormatter.java

```diff
@@ -5,14 +5,38 @@
 public abstract class StarDateFormatter {
 
         private static final String TIME_FORMAT_FOLDER_URL = "data/timeformat/";
+        private Path structurePath;
+        private String calenderName;
+        private String calenderDay;
+
 
         public static StarDateFormatter build(Path storyPath, String calendarName) {
             Path structurePath = storyPath.resolve(TIME_FORMAT_FOLDER_URL + calendarName + ".txt");
 
             StarDateFormatter formatter = null;
 
             return switch (calendarName){
-                case "iso8601" -> new StarDateEarthCalendar(structurePath, calendarName);
+                case "iso8601" -> new StarDateCustomCalendar(structurePath, calendarName, calenderDay) {
+                    @Override
+                    public String toString(StarDate timestamp, String formatName) {
+                        return null;
+                    }
+
+                    @Override
+                    public long toFutureTimestamp(StarDate currentTimestamp, long value, String unit) {
+                        return 0;
+                    }
+
+                    @Override
+                    public String getCalendarName() {
+                        return null;
+                    }
+
+                    @Override
+                    public String calenderDay() {
+                        return null;
+                    }
+                };
```

Comment:
why did you change this here? the StarDateEarthCalendar implements the working of the iso8601 calendar which is the calendar that most of us earthling uses

and the StarDateCustomCalendar you put here does nothing, none of the functionalities of the class were implemented here which at best will do nothing and at worst will crash the program